### PR TITLE
Docs Cloud and Concepts pages refactor

### PR DESCRIPTION
**Pull request by discdiver**
Reorganizes the Cloud and Concepts pages. Removes UI pages. Goal is better docs UX so that users can find what they are looking for move quickly.

Deploy preview: https://deploy-preview-9039--prefect-docs-preview.netlify.app/

### Checklist

- [ ] This pull request references any related issue by including "closes `<link to issue>`"
	- If no issue exists and your change is not a small fix, please [create an issue](https://github.com/PrefectHQ/prefect/issues/new/choose) first.
- [x] This pull request includes tests or only affects documentation.
- [x] This pull request includes a label categorizing the change e.g. `fix`, `feature`, `enhancement`, `docs`.

### CODE REVIEW

File: docs/concepts/index.md [hunk 1]
```diff
@@ -7,27 +7,10 @@ tags:
...
 ---
 
-# Concepts Overview
+# Explore Prefect concepts
 
-Getting started building and running workflows with Prefect doesn't require much more than a knowledge of Python and an intuitive understanding of "tasks" and "flows".  However, deploying and scheduling workflows as well as more advanced usage patterns do require a deeper understanding of the building blocks of Prefect.
+Learn how to develop, deploy, and host Prefect by exploring the concepts in the menu to the left.
 
-These guides are intended to provide the reader with a deeper understanding of how the system works and how it can be used to its full potential; in addition, these guides can be revisited as reference material as you learn more.
+Features specific to [Prefect Cloud](../cloud/) are in their own subheading.
```
Comment (cicdw) [inline docs/concepts/index.md:14]: this seems to be rendering as a heading for some reason?

Comment (discdiver) [reply docs/concepts/index.md:14]: fixed

**Comment by billpalombi** [comment]
This is great! Thank you @discdiver!

If we are revisiting organization, particularly of the concepts, I think we should consider:
- Removing the "Notifications" page entirely. Its much more like the UI pages than it is a concept.
- Moving the entire "Host" subsection of "Concepts" to be its own top level nav item. Neither "Server" nor "Database" are concepts.
- Move the "Database Settings" section of the "Settings" page to the "Database" page
- Move the "Logging Settings" section of the "Settings" page to the "Logging" page
- Make the "Settings" page something like "Profiles and configuration"

**Comment by billpalombi** [comment]
This PR appears to drop the artifacts concept page. Is that intentional?

**Comment by discdiver** [comment]
> This PR appears to drop the artifacts concept page. Is that intentional?

Nope. Good catch. Probably just accidentally cut the link. Will get it back in.

**Comment by discdiver** [comment]
> If we are revisiting organization, particularly of the concepts, I think we should consider:
> 
> * Removing the "Notifications" page entirely. Its much more like the UI pages than it is a concept.
> * Moving the entire "Host" subsection of "Concepts" to be its own top level nav item. Neither "Server" nor "Database" are concepts.
> * Moving the "Database Settings" section of the "Settings" page to the "Database" page
> * Moving the "Logging Settings" section of the "Settings" page to the "Logging" page
> * Moving the "Settings" page something like "Profiles and configuration"

Good thought @billpalombi! Implemented in the latest changes.

**Comment by billpalombi** [comment]
The "Profiles and Configuration" page looks good to me, but the "Host" page doesn't seem to have its child pages.

**Comment by discdiver** [comment]
> The "Profiles and Configuration" page looks good to me, but the "Host" page doesn't seem to have its child pages.

I folded database and server into the Host page itself - as serve is most of what would be on the host page and having a single subpage seemed suboptimal.  I can configure that differently if you like @billpalombi

**Comment by billpalombi** [comment]
> I folded database and server into the Host page itself

Thats much better than my idea. The host page has the "Getting Started" content on it for me, but I see that its rebuilding now so perhaps that will be fixed.

File: docs/tutorials/index.md [hunk 2]
```diff
@@ -36,3 +36,5 @@ If you've never used Prefect before, let's start by exploring the core concepts:
 If you have used Prefect 1 ("Prefect Core") and are familiar with Prefect workflows, we still recommend reading through these first steps, particularly [Run a flow within a flow](/tutorials/first-steps/#run-a-flow-within-a-flow). Prefect 2 flows and subflows offer significant new functionality.
 
 
+
+placeholder for Emil
```
Comment (billpalombi) [inline docs/tutorials/index.md:40]:
```suggestion
```

**Review by billpalombi — CHANGES_REQUESTED** [review]
Just need to remove the placeholder and this is good to go

**Review by billpalombi — APPROVED** [review]
Lets do it

**Pull request merged**